x-pack/metricbeat/module/azure/monitor_service.go

```diff
@@ -29,16 +29,18 @@ type MonitorService struct {
 const metricNameLimit = 20
 
 // NewService instantiates the Azure monitoring service
-func NewService(clientId string, clientSecret string, tenantId string, subscriptionId string) (*MonitorService, error) {
-	clientConfig := auth.NewClientCredentialsConfig(clientId, clientSecret, tenantId)
+func NewService(config Config) (*MonitorService, error) {
+	clientConfig := auth.NewClientCredentialsConfig(config.ClientId, config.ClientSecret, config.TenantId)
```

Comment:
Could this function be called with just `config` as input to avoid this unfolding and the extra assignments bellow?

Reply:
you mean the `auth.NewClientCredentialsConfig`, this is part of the azure sdk so we cannot pass the config we have.

Reply:
ah, I see 👍🏼